### PR TITLE
Recursive activation/deactivation module

### PR DESCRIPTION
**Pull request by Soldras**
add option --with-dependencies on module:activate and module:deactivate to recursively activate or deactivate linked module

### CODE REVIEW

File: core/lib/Thelia/Action/Module.php [hunk 1]
```diff
@@ -147,6 +153,61 @@ private function checkDeactivation($module)
...
+                foreach ($dependencies as $defMod) {
+                    $submodule = ModuleQuery::create()
+                        ->findOneByCode($defMod["code"]);
+                    if ($submodule && $submodule->getActivate() != BaseModule::IS_ACTIVATED) {
+                        $subevent = new ModuleToggleActivationEvent($submodule->getId());
+                        $subevent->setRecursive(true);
+                        $event->getDispatcher()->dispatch(TheliaEvents::MODULE_TOGGLE_ACTIVATION, $subevent);
+                    }
+                }
+            } catch (\Exception $ex) {
```
Comment (lunika) [inline core/lib/Thelia/Action/Module.php]: Why catching all exceptions and then throw it again ?

File: core/lib/Thelia/Action/Module.php [hunk 2]
```diff
@@ -147,6 +153,61 @@ private function checkDeactivation($module)
...
+                foreach ($dependencies as $defMod) {
+                    $submodule = ModuleQuery::create()
+                        ->findOneByCode($defMod["code"]);
+                    if ($submodule && $submodule->getActivate() == BaseModule::IS_ACTIVATED) {
+                        $subevent = new ModuleToggleActivationEvent($submodule->getId());
+                        $subevent->setRecursive(true);
+                        $event->getDispatcher()->dispatch(TheliaEvents::MODULE_TOGGLE_ACTIVATION, $subevent);
+                    }
+                }
+            } catch (\Exception $ex) {
```
Comment (lunika) [inline core/lib/Thelia/Action/Module.php]: same here

File: core/lib/Thelia/Action/Module.php (hunk 1 above)
Comment (Soldras) [inline core/lib/Thelia/Action/Module.php]: If you check these methode : 
*  private function checkDeactivation($module)
*  private function checkActivation($module)

There is the same Exception handler so I assume is needed

Comment (lunika) [inline core/lib/Thelia/Action/Module.php]: Well this is not really a reason to copy/paste crappy code.

**Comment by lovenunu** [comment]
you introduced a new issue on scrutinizer https://scrutinizer-ci.com/g/thelia/thelia/inspections/fa8e2077-97d3-4f8f-8991-6472e1d7d9b3/issues/files/core/lib/Thelia/Command/ModuleDeactivateCommand.php?status=new&orderField=path&order=asc

**Comment by lovenunu** [comment]
And now @Soldras, it's time to squash your commits !
ouch

**Comment by Soldras** [comment]
Cool I do it !

**Pull request merged**